api/src/services/dwc-service.ts

```diff
@@ -18,6 +18,39 @@ export class DwCService extends DBService {
     super(connection);
   }
 
+  /**
+   * Creates a set of all taxon IDs from the provided object
+   *
+   * @param {any} patchToPatch
+   * @return {*}  {Set<string>}
+   * @memberof DwCService
+   */
+  collectTaxonIDs(pathsToPatch: any): Set<string> {
+    const taxonSet = new Set<string>();
+    pathsToPatch.forEach((item: any) => {
+      taxonSet.add(item.value['taxonID']);
+    });
+    return taxonSet;
+  }
+
+  /**
+   * Calls elastic search taxonomy service with the given set of taxon codes
+   *
+   * @param {Set<string>} set A set of taxon IDs
+   * @return {*}  {Promise<any>}
+   * @memberof DwCService
+   */
+  async getEnrichedDataForSpeciesSet(set: Set<string>): Promise<any> {
```

Comment:
Can we make the `Promise<any>` more specific?

Reply:
Either some kind of `Record<string, whatevergetEnrichedDataForSpeciesCodeReturns>`

or maybe even a `new Map<string, whatevergetEnrichedDataForSpeciesCodeReturns>` (assuming Map is useful here, can't really remember what the pros/cons of Map are tbh)

Reply:
the difference I could see between record and map is that map can be iterated over. It also provides functions like `.has(key)` or `.size`. I'm put a map because it looks a little cleaner to add/ retrieve the data.